### PR TITLE
Convert weights to tensorflow protobuf file

### PR DESCRIPTION
**Pull request by jamesFerris**
Hi, this is my first time submitting a proper pull request, so please let me know if anything goes against best practices.
I've been trying implement yolo-v3 in tensorflows C api, which required converting it into a protobuf format.
The convert_weights_pb.py freezes the weights to a protobuf graph. I've modified demo.py to also accept the frozen graph.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
-# tensorflow-yolo-v3
+# tensorflow-yolo-v3 (fork)
```
Comment (mystic123) [inline README.md]: This should not be changed

File: README.md [hunk 2]
```diff
@@ -1,4 +1,8 @@
...
+
+Converts the yolov3 weights to a full protobuf graph, enabling its use in other apis (e.g. the tensorflow C api).
+
+[Original repo](https://github.com/mystic123/tensorflow-yolo-v3)
```
Comment (mystic123) [inline README.md]: Not needed

File: demo.py [hunk 3]
```diff
@@ -23,52 +24,91 @@
...
 
-tf.app.flags.DEFINE_float('conf_threshold', 0.5, 'Confidence threshold')
-tf.app.flags.DEFINE_float('iou_threshold', 0.4, 'IoU threshold')
+tf.app.flags.DEFINE_float(
+    'conf_threshold', 0.5, 'Confidence threshold')
+tf.app.flags.DEFINE_float(
+    'iou_threshold', 0.4, 'IoU threshold')
+
+tf.app.flags.DEFINE_float(
+    'gpu_memory_fraction', 0.8, 'Gpu memory fraction to use')
```
Comment (mystic123) [inline demo.py]: Is this flag required? Maybe rather set it to 1.0 by default

File: demo.py [hunk 4]
```diff
@@ -23,52 +24,91 @@
...
+
+        with tf.variable_scope('detector'):
+            detections = model(inputs, len(classes),
+                               data_format=FLAGS.data_format)
+
+        saver = tf.train.Saver(var_list=tf.global_variables(scope='detector'))
+
+        boxes = detections_boxes(detections)
+
+        tf_session = tf.Session(
```
Comment (mystic123) [inline demo.py]: I prefer using "with tf.Session() ..." rather than explicitly calling close() later. Please change

File: demo.py [hunk 5]
```diff
@@ -23,52 +24,91 @@
...
     classes = load_coco_names(FLAGS.class_names)
 
-    # placeholder for detector inputs
-    inputs = tf.placeholder(tf.float32, [1, FLAGS.size, FLAGS.size, 3])
+    if FLAGS.frozen_model:
 
-    with tf.variable_scope('detector'):
-        detections = model(inputs, len(classes),
-                           data_format=FLAGS.data_format)
+        t0 = time.time()
```
Comment (mystic123) [inline demo.py]: Maybe wrap this 2 if branches into functions which returns inputs and boxes?

File: README.md [hunk 6]
```diff
@@ -16,17 +20,50 @@ To run demo type this in the command line:
...
-- Bug fixes
\ No newline at end of file
+    1. Download binary file with desired weights: 
+        1. Full weights: `wget https://pjreddie.com/media/files/yolov3.weights`
+        1. Tiny weights: `wget https://pjreddie.com/media/files/yolov3-tiny.weights` 
+    2. Run `python ./convert_weights.py` and `python ./convert_weights_pb.py`        
+3. Run `python ./demo.py --input_img <path-to-image> --output_img <name-of-output-image> --frozen_model <path-to-frozen-model>`
+
+
+####Optional Flags
```
Comment (mystic123) [inline README.md]: Thanks for this, this was really needed. Good job :)

File: README.md (hunk 1 above)
Comment (jamesFerris) [reply README.md]: Thanks. Like I said - I haven't done a proper pull request before, so I wasn't sure of the proper etiquette.

File: README.md (hunk 2 above)
Comment (jamesFerris) [reply README.md]: Removed.

File: demo.py (hunk 3 above)
Comment (jamesFerris) [reply demo.py]: I put that flag in just about everything, as I'm often trying to run multiple graphs concurrently. Should definitely default to 1.0. Let me know if you don't think its necessary.

File: demo.py (hunk 4 above)
Comment (jamesFerris) [reply demo.py]: Done.

File: demo.py (hunk 5 above)
Comment (jamesFerris) [reply demo.py]: Made a function to wrap this, but I'm not so certain about how I handled it. Your input would be welcomed.

Comment (jamesFerris) [reply demo.py]: I think I was severely overthinking this. I've simplified things a bit. Let me know if it is what you had in mind.

**Pull request merged**